### PR TITLE
Pcap HTTP Extractor Script

### PR DESCRIPTION
**Pull request by funerr**
## Status
Ready

## Related Issues
[https://github.com/demisto/etc/issues/14677](https://github.com/demisto/etc/issues/14677
)
## Description
An automated script that allows to parse and extract http flows from a pcap/pcapng file.

## Screenshots
image

## Related PRs
--
branch | PR
------ | ------

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - I think not.

## Must have
- [x] Tests (More tests were done via unittests, as a preamble to the future of scripts, ask @hirshbergshachar )
- [ ] Documentation (none because it's a script)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.

## Additional changes
Nothing out of the script itself.

## External dependencies:
- tshark (wireshark, GNU GPL V2 - https://github.com/demisto/etc/issues/14677#issuecomment-453987702) 
- pyshark (MIT)

### CODE REVIEW

File: Scripts/script-PcapHTTPExtractor.yml [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+  defaultValue: "0"
+outputs:
+- contextPath: PcapHTTPExtractor.Flows
+  description: Flows extracted from the pcap file.
+  type: string
+scripttarget: 0
+dockerimage: demisto/pcap-http-extractor:1.0.0.98
+releaseNotes: "Adding a new script that allows to extract http flows from pcap/pcapng files."
+tests:
+- blockip_test_playbook
```
Comment (glicht) [inline Scripts/script-PcapHTTPExtractor.yml]: Looks like a typo. Should have here the Test-PcapHTTPEtractor playbook

File: TestPlaybooks/script-PcapHTTPExtractor.yml [hunk 2]
```diff
@@ -0,0 +1,239 @@
+id: Test-PcapHTTPExtractor
```
Comment (glicht) [inline TestPlaybooks/script-PcapHTTPExtractor.yml]: If its a playbook file should be named: playbook-<Name>.yml

File: Scripts/script-PcapHTTPExtractor.yml [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+          entry_file_name = demisto.get(entry, 'File')
+          is_correct_file = file_name.lower() == entry_file_name.lower()
+          has_correct_extension = file_has_extension(file_name, extensions) if extensions else True
+
+          if is_correct_file and has_correct_extension:
+              found_entry_id = entry['ID']
+              break
+
+      if not found_entry_id:
+          return_error(demisto.args()['pcapFileName'] + ' no such file in war room')
```
Comment (glicht) [inline Scripts/script-PcapHTTPExtractor.yml]: No need to return an error. An empty table will be fine

File: Scripts/script-PcapHTTPExtractor.yml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+          return_error("Error occurred while parsing output from command. Exception info:\n" \
+                       + str(ex) + "\n\nTrace:\n" + str(traceback.format_exc()))
+type: python
+tags:
+  - pcap
+  - http
+  - pcapng
+  - Utility
+comment: An automated script that allows to parse and extract http flows from a pcap/pcapng file.
```
Comment (meirwah) [inline Scripts/script-PcapHTTPExtractor.yml]: drop the `An automated script that ` part

File: Scripts/script-PcapHTTPExtractor.yml [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+
+          PCAP_FILE = demisto.args()["pcapFileName"]
+          entry_id = find_entry_id_by_name(PCAP_FILE, [".pcap", ".cap", ".pcapng"])
+      elif 'entryID' in demisto.args():
+          entry_id = demisto.args()["entryID"]
+      else:
+          return_error('You must set pcapFileName or entryID when executing PcapHTTPExtract script')
+
+      res = demisto.executeCommand('getFilePath', {'id': entry_id})
+      if res[0]['Type'] == entryTypes['error']:
```
Comment (meirwah) [inline Scripts/script-PcapHTTPExtractor.yml]: verify res has len of > 0 , so no out of bound error

File: Scripts/script-PcapHTTPExtractor.yml [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+      except Exception as ex:
+          import traceback
+
+          return_error("Error occurred while parsing output from command. Exception info:\n" \
+                       + str(ex) + "\n\nTrace:\n" + str(traceback.format_exc()))
+type: python
+tags:
+  - pcap
+  - http
```
[the remark this answers is not in the document]
Comment (funerr) [reply Scripts/script-PcapHTTPExtractor.yml]: which tag?

**Comment by funerr** [comment]
> Don't commit the file: Scripts/script-PcapHTTPExtractor.py . You should commit only the yaml file (we will be supporting this in the future)

Removed.

File: Scripts/script-PcapHTTPExtractor.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (meirwah) [reply Scripts/script-PcapHTTPExtractor.yml]: http

**Comment by glicht** [comment]
See my comment at: https://github.com/demisto/etc/issues/14677#issuecomment-455908969

**Comment by meirwah** [comment]
@glicht can you review changes?

**Comment by glicht** [comment]
Looks good. Nice job.

**Pull request merged**